ape_solidity/compiler.py

```diff
@@ -666,8 +670,22 @@ def _get_pragma_spec(self, path: Path) -> Optional[NpmSpec]:
 
         compiler_version = pragma_spec.select(self.available_versions)
         if compiler_version:
-            solcx.install_solc(compiler_version, show_progress=False)
+            install_solc(compiler_version, show_progress=False)
         else:
+            # Attempt to use the best-installed version.
+            # NOTE: Use the oldest version available for maximum compatibility.
```

Comment:
Why use the oldest matching version versus the newest?

Reply:
@wakamex wrote it was for maximum compatibility but tbh i was wondering that too because it is kind of the opposite of what we do above and our plugin is supposed to use the best-smallest version it can find anyway.

Reply:
the idea was to maximize compatibility. my assumption is not having access to the internet to download solidity versions is a temporary (airplane) environment, where you just want things to work to go on with your minimal work, not to optimize for the latest and greatest (which you'll do once you get back to the real world).

but this is a judgement call, could go the other way. I could even see it being a toggle (like `--newest`).

not even sure how likely this is, but I imagine someone who wrote `^pragma 0.8` could have done so any time betwen Dec 2020 and now, with possible breaking changes coming afterward? or at least some undesired effects?

Reply:
One could argue that the author of `^pragma 0.8` is asking to use the latest, whether that is always a good idea or not is really debatable but the author is saying to do it so we should maybe honor such request and use the biggest version we have that works. If it fails, it is a sign that perhaps that author should restrain their version more if the user of Ape should configure the Solidity version directly, both of which seem like safe ideas to do anyway if you ask me.

Reply:
It should use the latest _installed_ version in airplane mode, with the note that it's not able to determine if newer versions exist to see if can install something newer

Reply:
I changed to the biggest version, but we can continue chatting if needed!

Reply:
> It should use the latest _installed_ version in airplane mode, with the note that it's not able to determine if newer versions exist to see if can install something newer

Agreed, that is what I meant as well, adjusted here [fccbd07](https://github.com/ApeWorX/ape-solidity/pull/114/commits/fccbd07c7a5d20c02336f9999ad7c308144be620)